internal/pkg/model/relation.go

```diff
@@ -0,0 +1,91 @@
+package model
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+const (
+	VariablesForRelType = RelationType(`variablesFor`)
+)
+
+type RelationType string
+
+func (t RelationType) String() string {
+	return string(t)
+}
+
+func (t RelationType) Type() RelationType {
+	return t
+}
+
+func newEmptyRelation(t RelationType) (Relation, error) {
+	switch t {
+	case VariablesForRelType:
+		return &VariablesForRelation{RelationType: VariablesForRelType}, nil
+	default:
+		return nil, fmt.Errorf(`unexpected RelationType "%s"`, t)
+	}
+}
+
+// Relation between objects, eg. config <-> config.
+type Relation interface {
+	String() string
+	Type() RelationType
+	TargetKey(source Key) (Key, error) // source is where the relation is defined, target is other side
+	ParentKey(source Key) (Key, error) // if relation type is parent <-> child, then parent key is returned, otherwise nil
+}
+
+type Relations []Relation
+
+func (v *Relations) UnmarshalJSON(data []byte) error {
```

Comment:
`Relation` moze mat rozne polozky, ... zatial je implementovana iba `VariablesForRelation` ale rata sa s dalsimi.
 Kazda tato implementacia je nejaka struktura, ktora implementuje interface `Relation`.
 Zaroven sa tieto `Relations` ukladaju do `manifest.json`.

 `Relations` je teda pole struktur rozneho typu so spolocnym interface.
 
 Pri ukladani to nie je problem, ... lebo kazda struktura ma cez `tags` definovane ako sa ma previest do JSONu.
 
 No je potrebne vyriesit nacitavanie, ... aby sa z JSONu vytvorila struktura spravneho typu.
 
 Na to sluzi kluc `type` (vid testy) a tento kod, kedze toto nevie Go spravit automaticky.

Reply:
Read more: https://mariadesouza.com/2017/09/07/custom-unmarshal-json-in-golang/